src/new/wallet.js

```diff
@@ -392,7 +392,7 @@ class HathorWallet extends EventEmitter {
     const xpub = HDPublicKey(accessData.xpubkey);
 
     // deserialize P2SHSignature for all signatures
-    const p2shSignatures = signatures.map(sig => P2SHSignature.deserialize(sig));
+    const p2shSignatures = signatures.sort().map(sig => P2SHSignature.deserialize(sig));
```

Comment:
I feel we should have a comment here explaining why this work. The signatures must be sorted in the same order as the xpub (as you said in the PR description). The `signatures` array have a P2SHSignature object, which has the xpubkey, then sorting this array will be the same as sorting the signatures in the xpub order

Reply:
Done in [46a30f1](https://github.com/HathorNetwork/hathor-wallet-lib/pull/369/commits/46a30f1da8c379b94475097b8fb075d250476ea2)